spec/takeaway_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'takeaway'
+require 'basket'
+require 'text'
+require 'pry'
+
+describe Takeaway do
+
+  let(:menu) { double(:menu) }
+  let(:current_basket) { double(:current_basket) }
+  let(:sms) { double(:sms) }
+  let(:takeaway) { Takeaway.new(menu, current_basket, sms) }
+
+  context 'Describe #menu:' do
+    before do
+      allow(menu).to receive(:view).and_return("menu")
+    end
+    it 'prints the menu' do
+      expect(takeaway.menu).to eq "menu"
+    end
+  end
+
+  context 'Describe #order:' do
+    before do
+      allow(menu).to receive_message_chain(:list, :has_key?).and_return(false)
```

Comment:
I think you're mocking a little too far here — `list` is a method on `Menu` which makes sense to mock`, but `has_key?` starts to be 'knowing too much' about the `#order` method. What if we changed how `#order` achieved its results? We'd have to change this test, even though it all still works :)

A better solution:

```ruby
allow(menu).to receive_message_chain(:list).and_return({"Boiled Rice" => 4})
```